core/src/main/java/com/linecorp/armeria/server/file/FileSystemHttpVfs.java

```diff
@@ -98,13 +108,13 @@ static HttpFile build(HttpFileBuilder builder,
                           Clock clock,
                           String pathOrUri,
                           @Nullable String contentEncoding,
-                          HttpHeaders additionalHeaders) {
+                          HttpHeaders additionalHeaders, MediaTypeResolver mediaTypeResolver) {
 
         builder.autoDetectedContentType(false);
         builder.clock(clock);
         builder.setHeaders(additionalHeaders);
 
-        final MediaType contentType = MimeTypeUtil.guessFromPath(pathOrUri, contentEncoding);
+        final @Nullable MediaType contentType = mediaTypeResolver.guessFromPath(pathOrUri, contentEncoding);
```

Comment:
nit; I don't think we use `@Nullable` for local variables.

Sorry this isn't documented somewhere 😅 https://github.com/line/armeria/pull/3806#issuecomment-913297854

Reply:
I'll remove `@Nullable` for local variables. Thanks for the advice.